Maple2.Server.Game/PacketHandlers/ItemUseHandler.cs

```diff
@@ -106,4 +117,91 @@ public class ItemUseHandler : PacketHandler<GameSession> {
+            session.Send(NoticePacket.MessageBox(new InterfaceText(StringCode.s_couple_effect_error_openbox_myself_char)));
+            return;
+        }
+
+        using GameStorage.Request db = session.GameStorage.Context();
+        CharacterInfo? receiverCharacterInfo = db.GetCharacterInfo(targetUser);
+        if (receiverCharacterInfo == default) {
+            session.Send(NoticePacket.MessageBox(new InterfaceText(StringCode.s_couple_effect_error_openbox_charname)));
+            return;
+        }
+
+        if (receiverCharacterInfo.AccountId == session.Player.Value.Character.AccountId) {
+            session.Send(NoticePacket.MessageBox(new InterfaceText(StringCode.s_couple_effect_error_openbox_myself_account)));
+            return;
+        }
+
+        int[] buddyBadgeBoxParams = item.Metadata.Function?.Parameters.Split(',').Select(int.Parse).ToArray() ?? Array.Empty<int>();
+        if (buddyBadgeBoxParams.Length != 2) {
+            session.Send(NoticePacket.MessageBox(new InterfaceText(StringCode.s_couple_effect_error_openbox_unknown)));
+            Logger.Error("Invalid buddy badge box parameters: {Parameters}", item.Metadata.Function?.Parameters);
+            return;
+        }
+
+        if (!ItemMetadata.TryGet(buddyBadgeBoxParams[0], out ItemMetadata? itemMetadata)) {
+            session.Send(NoticePacket.MessageBox(new InterfaceText(StringCode.s_couple_effect_error_openbox_unknown)));
+            Logger.Error("Unknown buddy badge box item Id: {Parameters}", buddyBadgeBoxParams[0]);
+            return;
+        }
+
+        var selfBadge = new Item(itemMetadata, buddyBadgeBoxParams[1]) {
+            CoupleInfo = new ItemCoupleInfo(receiverCharacterInfo.CharacterId, receiverCharacterInfo.Name, true),
+        };
+
+        if (!session.Item.Inventory.CanAdd(selfBadge)) {
+            session.Send(NoticePacket.MessageBox(new InterfaceText(StringCode.s_gem_error_inventory_full)));
+            return;
+        }
+
+        if (!session.Item.Inventory.Consume(item.Uid, 1)) {
+            Logger.Error("Failed to use buddy badge box: {ItemUid}", item.Uid);
+            return;
+        }
+
+        var receiverMail = new Mail() {
+            ReceiverId = receiverCharacterInfo.CharacterId,
+            Type = MailType.System,
+            ContentArgs = new[] {
+                ("str", $"{session.Player.Value.Character.Name}"),
+            },
+        };
+
+        receiverMail.SetTitle(StringCode.s_couple_effect_mail_title_receiver);
+        receiverMail.SetContent(StringCode.s_couple_effect_mail_content_receiver);
+        receiverMail.SetSenderName(StringCode.s_couple_effect_mail_sender);
+
+        receiverMail = db.CreateMail(receiverMail);
+        if (receiverMail == null) {
+            session.Send(NoticePacket.MessageBox(new InterfaceText(StringCode.s_couple_effect_error_openbox_unknown)));
+            return;
+        }
+
+        using GameStorage.Request itemDb = session.GameStorage.Context();
```

Comment:
You don't need a separate db connection, just use the same one from above. (Actually just realized this is unused, so remove it)